concurrentmap.go

```diff
@@ -45,3 +45,51 @@ func (m *ConcurrentMap) Keys() []string {
 	}
 	return keys
 }
+
+type ConcurrentIntMap struct {
+	mp    map[interface{}]*SafeInt
+	mutex sync.RWMutex
+}
+
+func NewConcurrentIntMap() *ConcurrentIntMap {
+	return &ConcurrentIntMap{
+		mp: map[interface{}]*SafeInt{},
+	}
+}
+
+func (m *ConcurrentIntMap) Create(key interface{}) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	m.mp[key] = NewSafeInt(0)
+}
+
+func (m *ConcurrentIntMap) Increment(key interface{}) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	m.mp[key].Increment()
+}
+
+func (m *ConcurrentIntMap) Get(key interface{}) (*SafeInt, bool) {
+	m.mutex.RLock()
+	defer m.mutex.RUnlock()
+	v, ok := m.mp[key]
+	return v, ok
+}
+
+func (m *ConcurrentIntMap) GetAll() map[interface{}]*SafeInt {
+	m.mutex.RLock()
+	defer m.mutex.RUnlock()
+	return m.mp
```

Comment:
This function won't be thread-safe since calling functions can modify the map from outside.